_gsocblogs/2022/blog_ROOT_ZifengLuo.md

```diff
@@ -28,86 +30,136 @@ Based on the data types supported by TTree, the project is divided into 7 parts
    </tr>
    <tr>
       <td >Convert TTree containing simple variables </td>
-      <td>int, float, bool, ...</td>
-      <td>Completed <a href="https://github.com/luozf14/TTreeToRNTuple/blob/main/src/ConverterSimple.cxx">[Code]</a></td>
+      <td><code>int</code>, <code>float</code>, <code>bool</code>, ...</td>
+      <td>Completed. Types are one-to-one mapped (e.g., <code>int</code>-><code>int</code>).</td>
    </tr>
    <tr>
       <td rowspan="3">Convert TTree containing arrays</td>
-      <td>Fixed-length array</td>
-      <td rowspan="2">Completed <a href="https://github.com/luozf14/TTreeToRNTuple/blob/main/src/ConverterArray.cxx">[Code]</a></td>
+      <td>Fixed-length 1D array</td>
+      <td>Completed. Mapped to <code>std::array&lt;T, N></code> (e.g., <code>int myArray[10]</code>-><code>std::array&lt;int,10></code>)</td>
    </tr>
    <tr>
-      <td>Variable-sized array</td>
+      <td>Variable-sized 1D array</td>
+      <td>Completed. Mapped to <code>std::vector&lt;T></code> (e.g., <code>int myArray[N]</code>-><code>std::vector&lt;int></code>)</td>
    </tr>
    <tr>
-      <td>Multidimensional array (e.g., int myArry[16][8])</td>
-      <td>Need RNTuple feature support</td>
+      <td>Multidimensional array (e.g., <code>int myArray[16][8]</code>)</td>
+      <td>Depending on if the size is variable, currently it is mapped to 1D <code>std::array&lt;T, N></code> (fixed-size) or <code>std::vector&lt;T></code> (variable-size). In order to map it to corresponding dimensional vector, native RNTuple feature support is needed.</td>
    </tr>
    <tr>
       <td rowspan="5">Convert TTree containing STL types and collections </td>
-      <td>std::string</td>
-      <td rowspan="5">Completed <a href="https://github.com/luozf14/TTreeToRNTuple/blob/main/src/ConverterSTLContainer.cxx">[Code]</a></td>
+      <td><code>std::string</code></td>
+      <td rowspan="5">Completed. All types are one-to-one mapped except types that are not supported by RNTuple I/O. For example, <code>std::set&lt;T></code> will cause program crash. However, conversion of TTree containing branches of <code>ROOT::RVec&lt;T></code> is not stable in current stage (see section <b>Known issues and Future work</b>).</td>
```

Comment:
```suggestion
      <td rowspan="5">Completed. All types are one-to-one mapped except types that are not supported by RNTuple I/O. For example, <code>std::set&lt;T></code> will not work. However, conversion of TTree containing branches of <code>ROOT::RVec&lt;T></code> is not stable in current stage (see section <b>Known issues and Future work</b>).</td>
```
We could actually handle the exception thrown in `RFieldBase::Create()`, but that is beyond the GSoC project goals.